lib/src/config_util.cc

```diff
@@ -62,4 +63,33 @@ namespace hocon {
         return result;
     }
 
+    string render_string_unquoted_if_possible(string const& s) {
+        // this can quote unnecessarily as long as it never fails to quote when
+        // necessary
+        if (s.empty()) {
+            return render_json_string(s);
+        }
+
+        // if it starts with a hyphen or number, we have to quote
+        // to ensure we end up with a string and not a number
+        char first = s[0];
+        if (isdigit(first) || first == '-') {
+            return render_json_string(s);
+        }
+
+        if (boost::starts_with(s, "include") || boost::starts_with(s, "true") || boost::starts_with(s, "false") ||
+                boost::starts_with(s, "null") || boost::starts_with(s, "//")) {
+            return render_json_string(s);
+        }
+
+        // only unquote if it's pure alphanumeric
+        for (char c : s) {
+            if (!(isalpha(c) || isdigit(c) || c == '-')) {
```

Comment:
I had to stare at this for a while too. It can unquote the string so long as every character is one of those three things (the first char is an exception, but we already checked for that). But if any of them is not those things, we need to put quotes on it, which is what render_json_string() does.